### PR TITLE
GitMate plugins project idea

### PR DESCRIPTION
**Pull request by rohanrk**
Ref: https://github.com/coala/projects/issues/488

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: _projects/gitmate_plugins.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+name: "Implement GitMate Plugins for coala"
+desc: "GitMate.io helps by automating tasks on GitHub and GitLab. There are new plugins that are specific to coala that should be implemented"
+requirements:
+  - "The participant should have at least one patch to [GitMate](https://gitlab.com/gitmate/open-source/gitmate-2) accepted."
+mentors:
```
Comment (jayvdb) [inline _projects/gitmate_plugins.md]: tentatively put in `sils` and `nkprince007`

File: _projects/gitmate_plugins.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+[GitMate.io](https://gitmate.io/) is an automation tool for developers. While it works well, there are a number of plugins that could automate tasks to save maintainer time (for example automatically close PRs from master). 
+
+[IGitt](https://igitt.gitmate.io/index.html) is a Git hoster abstraction for python. It's an extensive framework that will help implement GitMate's plugins. 
+
+To get started on GitMate, please follow [our guide](http://docs.gitmate.io/Developers/Newcomers_Guide/).
+
+#### Plugin Ideas
+
+- Reject Pull Requests from master. Should include instructions on how to fix their branch, fix their master, and resubmit a PR.
```
Comment (jayvdb) [inline _projects/gitmate_plugins.md]: cut lines at 80cpl

**Comment by rohanrk** [comment]
Done @jayvdb

**Comment by nkprince007** [comment]
From the development perspective, I don't think building the plugins listed in the structure will take up the whole period of GSoC. Typically, implementing all of the stated ones can happen in less than a month. Please propose more ideas for plugins or enhance the existing ones so that they can be readily used by a lot of other organizations.

Also, the configuration part, as suggested by @Makman2 can be a good addition to GitMate within this project.

**Comment by rohanrk** [comment]
This and https://github.com/coala/projects/issues/487 work well as one GitMate GSOC project. 

https://gitlab.com/gitmate/open-source/gitmate-2/issues/363 has some more details on @Makman2's configuration idea.

I'll edit this.

**Comment by jayvdb** [comment]
@gitmate-bot rebase

File: _projects/gitmate_plugins.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+tags:
+  - IGitt
+  - GitMate
+collaborating_projects:
+  - coala
+  - GitMate
+  - IGitt
+---
+
+[GitMate.io](https://gitmate.io/) is an automation tool for developers. While it works well, there are a number of plugins that could automate tasks to save maintainer time (for example automatically close PRs from master).
```
Comment (jayvdb) [inline _projects/gitmate_plugins.md]: wrap lines at 80cpl

File: _projects/gitmate_plugins.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+[IGitt](https://igitt.gitmate.io/index.html) is a Git hoster abstraction 
+for python. It's an extensive framework that will help implement GitMate's plugins.
+
+To get started on GitMate, please follow 
+[our guide](http://docs.gitmate.io/Developers/Newcomers_Guide/).
+
+#### Plugin Ideas
+
+- Reject Pull Requests from master. Should include 
+instructions on how to fix their branch, fix their master, and resubmit a PR.
```
Comment (jayvdb) [inline _projects/gitmate_plugins.md]: indent properly

File: _projects/gitmate_plugins.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  - Similarly marking PRs as WIP if CI tests fail
+- Auto-merger created to integrate rebase and ff plugins
+
+#### Milestones
+
+##### PREPARATION/BONDING
+- GitMate should automatically reject PRs from master
+
+##### Coding Phase 1
+- Fixes vs Closes automated checking should be implemented
```
Comment (jayvdb) [inline _projects/gitmate_plugins.md]: Also add a 'recheck' comment, for cases where the problem was unrelated to the PR, and has been fixed, just like has happened with your PR here.

File: _projects/gitmate_plugins.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+- A squash command to fix a PR with many messy commits. 
+Should link to the coala's git documentation for future reference
+- Automatically marking WIP if GitMate's coala plugin finds defects.
+  - Similarly marking PRs as WIP if CI tests fail
+- Auto-merger created to integrate rebase and ff plugins
+
+#### Milestones
+
+##### PREPARATION/BONDING
+- GitMate should automatically reject PRs from master
```
Comment (jayvdb) [inline _projects/gitmate_plugins.md]: ".. from master of forks"

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/513/commits/c4adfa9c9aaa99bcfdbe56586defed7086cdaa2c

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**